### ISSUE TITLE
Initial table implementation

### ISSUE THREAD

**Issue by atmgrifter00** [ni/nimble#840]
## 📌 User Story

This user story covers the creation of the `nimble-table` component, and any other components needed to render a set of data as outlined in the data API HLD (spec reference coming).

### Accessibility
This basic table should have appropriate aria labels to meet accessibility standards
This version of the table will use the table role
https://w3c.github.io/aria-practices/examples/table/table.html
https://w3c.github.io/aria-practices/#table

### Design
This table should implement the basic visual design:
 - #883 

### Platform support
The basic table should have versions available for Angular and Blazor

### Other functionality
If the data of the table updates, the table should update
- This behavior allows existing workflows of filtering and searching to work.

### Related PRs
1. #902 
1. #914 
1. #921 
1. #928
1. #931
1. #932 
2. #945 
3. #969

**Comment by nate-ni** [comment]
@RickA-NI, can you verify that the table implementation matches the visual design spec?
You can see an example of the table at: https://nimble.ni.dev/storybook/?path=/story/table--table

@mollykreis, do we need to validate the Angular and Blazor (when ready) implementations as well or is the code shared enough that they will be identical?

**Comment by mollykreis** [comment]
> do we need to validate the Angular and Blazor (when ready) implementations as well or is the code shared enough that they will be identical?

@nate-ni, there is no need to validate the styling in Angular and Blazor. All the styling is shared.

**Comment by RickA-NI** [comment]
The Light skin and Dark skin look correct the something is off on the color skin the lighter background color for the rows and headers is not showing up all I'm seeing is text on the dark green.

**Comment by mollykreis** [comment]
@RickA-NI, we were unsure the correct way to style the Color theme. We were confused because the dividers between the rows was speced to be ForestGreen@75% while the rows themselves are ForestGreen@100%. This means that the dividers are only visible on some backgrounds. Could you clarify what color the dividers should be on the Color theme? ForestGreen@75% layered on top of some other color (that at least I couldn't find in the XD document)?

If the answer isn't straight forward, we can follow up in person.

**Comment by RickA-NI** [comment]
@mollykreis looks like that was due to poor design on my part. I've updated the color page of the spec to match how the other color controls are styled. It should work properly now. If there's questions let me know.

**Comment by nate-ni** [comment]
- #969 resolves the open items to get this issue resolved.
